cmd/app.go

```diff
@@ -687,10 +688,29 @@ func createNamespace(c *cliclient.MasterClient, n string) error {
 			ProjectID: c.UserConfig.Project,
 		}
 
-		_, err = c.ClusterClient.Namespace.Create(newNamespace)
+		ns, err := c.ClusterClient.Namespace.Create(newNamespace)
 		if err != nil {
 			return err
 		}
+
+		var timeout int
+		for {
+			logrus.Debug(fmt.Sprintf("Namespace create wait - Name: %s, State: %s, Transitioning: %s", ns.Name, ns.State, ns.Transitioning))
+			if timeout == 60 {
+				return fmt.Errorf("timed out waiting for new namespace %s", ns.Name)
+			}
+			ns, err = c.ClusterClient.Namespace.ByID(ns.ID)
+			if err != nil {
+				return err
+			}
+
+			if ns.State == "active" {
+				break
+			}
+
+			time.Sleep(1 * time.Second)
```

Comment:
My opinion: sleeping a second is too long and waiting an entire minute before erroring out is too long.  Half a second and 30 total would be slightly better?

Other than that, looks good

Reply:
Updated to half a second and timeout at 30 seconds